src/client/pages/SignIn.tsx

```diff
@@ -145,7 +146,23 @@ export const SignIn = ({
 	shortRequestId,
 	usePasscodeSignIn = false,
 	hideSocialButtons = false,
+	focusPasswordField = false,
 }: SignInProps) => {
+	const [currentEmail, setCurrentEmail] = React.useState(email);
+
+	// autofocus the password input field when the page loads if it exists
+	// and the focusPasswordField flag is set to true and a default email exists
+	useEffect(() => {
+		if (typeof window !== 'undefined' && focusPasswordField && email) {
+			const passwordInput: HTMLInputElement | null =
+				window.document.querySelector('input[name="password"]');
+
+			if (passwordInput) {
+				passwordInput.focus();
+			}
+		}
+	}, [focusPasswordField, email]);
```

Comment:
Would [`autofocus`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/autofocus) work here?

For some reason my screen reader isn't announcing the "password" field when its getting `.focus()`ed, not sure if thats a side effect from it being called by `useEffect` or if im just using the screen reader wrong, but I wonder if using the browsers built in autofocus it might work better.

Reply:
I'll refactor to use `autofocus` anyway!

As it avoids a `useEffect` and any client side code, as the `autofocus` attribute can be applied on the server side render of the page!